app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+
+  def post_update_counter
+    user = User.find(author_id)
+    user.update_attribute(:posts_count, user.posts_count + 1)
+  end
```

Comment:
[OPTIONAL] You can use one of the Rails Active Callbacks - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private

```
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    # logic here
  end
```